src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PossiblyDeclareAsNullable
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer :
+        DiagnosticAnalyzer
```

Comment:
Derive from AbstractCodeStyleDiagnosticAnalyzer or AbstractBuiltInCodeStyleDiagnosticAnalyzer and you can cut out a lot of this boilerplate.  taht's what most IDE analyzers do.